lib/Partitioner/Partitioner.cpp

```diff
@@ -468,6 +468,7 @@ void Partitioner::doPartitioning(Function *F, NodeToFunctionMap &mapping) {
     }
 
     // Link subF to its parents.
+    std::set<Function *> parents;
```

Comment:
Assuming `parents` usually has a small number of Functions, you could consider using `llvm::SmallSet` here instead.